### PR TITLE
Some other code now uses ai status display list

### PR DESCRIPTION
**Pull request by optimumtact**

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -632,13 +632,12 @@
 		return //won't work if dead
 	var/list/ai_emotions = list("Very Happy", "Happy", "Neutral", "Unsure", "Confused", "Sad", "BSOD", "Blank", "Problems?", "Awesome", "Facepalm", "Friend Computer", "Dorfy", "Blue Glow", "Red Glow")
 	var/emote = input("Please, select a status!", "AI Status", null, null) in ai_emotions
-	for (var/obj/machinery/M in GLOB.machines) //change status
+	for (var/obj/machinery/M in GLOB.ai_status_displays) //change status of displays
 		if(istype(M, /obj/machinery/ai_status_display))
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: These aren't in ```ai_status_displays```

Comment (optimumtact) [inline code/modules/mob/living/silicon/ai/ai.dm]: Heh, this means the previous PR was never working in the first place

Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: Indeed it does, yes.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -632,13 +632,12 @@
 		return //won't work if dead
 	var/list/ai_emotions = list("Very Happy", "Happy", "Neutral", "Unsure", "Confused", "Sad", "BSOD", "Blank", "Problems?", "Awesome", "Facepalm", "Friend Computer", "Dorfy", "Blue Glow", "Red Glow")
 	var/emote = input("Please, select a status!", "AI Status", null, null) in ai_emotions
-	for (var/obj/machinery/M in GLOB.machines) //change status
+	for (var/obj/machinery/M in GLOB.ai_status_displays) //change status of displays
```
Comment (lzimann) [inline code/modules/mob/living/silicon/ai/ai.dm]: Remove the machinery typecast here, it's not being used for anything besides an extra typecheck.

**Pull request merged**